newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/EventsFunctionPropertiesEditor.js

```diff
@@ -237,69 +280,223 @@ export default class EventsFunctionPropertiesEditor extends React.Component<
+                  floatingLabelText={
+                    <Trans>Description, displayed in editor</Trans>
+                  }
+                  fullWidth
+                  multiline
+                  value={
+                    getterFunction
+                      ? 'Change ' + getterFunction.getDescription()
+                      : ''
+                  }
+                  onChange={text => {}}
+                />
+              ) : (
+                <SemiControlledTextField
+                  commitOnBlur
+                  floatingLabelText={
+                    type === gd.EventsFunction.ExpressionAndCondition ||
+                    type === gd.EventsFunction.StringExpressionAndCondition ? (
+                      <Trans>
+                        Description, displayed in editor (automatically prefixed
+                        by "Compare" or "Return")
+                      </Trans>
+                    ) : (
+                      <Trans>Description, displayed in editor</Trans>
+                    )
+                  }
+                  translatableHintText={getDescriptionHintText(type)}
+                  fullWidth
+                  multiline
+                  value={eventsFunction.getDescription()}
+                  onChange={text => {
+                    eventsFunction.setDescription(text);
+                    if (onConfigurationUpdated) onConfigurationUpdated();
+                    this.forceUpdate();
+                  }}
+                />
+              )}
             </Line>
             <Line noMargin>
-              {type === gd.EventsFunction.Action ||
-              type === gd.EventsFunction.Condition ? (
+              {type === gd.EventsFunction.ActionWithOperator ? (
                 <SemiControlledTextField
+                  disabled
                   commitOnBlur
                   floatingLabelText={<Trans>Sentence in Events Sheet</Trans>}
+                  fullWidth
+                  value={
+                    getterFunction
+                      ? 'Change ' +
+                        getterFunction.getSentence() +
+                        ' of _PARAM0_'
+                      : ''
+                  }
+                  onChange={text => {}}
+                />
+              ) : type === gd.EventsFunction.Action ||
+                type === gd.EventsFunction.Condition ||
+                type === gd.EventsFunction.ExpressionAndCondition ||
+                type === gd.EventsFunction.StringExpressionAndCondition ? (
+                <SemiControlledTextField
+                  commitOnBlur
+                  floatingLabelText={
+                    eventsBasedBehavior &&
+                    (type === gd.EventsFunction.ExpressionAndCondition ||
+                      type ===
+                        gd.EventsFunction.StringExpressionAndCondition) ? (
+                      <Trans>
+                        Sentence in Events Sheet (automatically suffixed by "of
```

Comment:
That could even be:
```suggestion
                        Name in Events Sheet (automatically suffixed by "of
```

Reply:
(because it's a name, not really a sentence anymore)

Reply:
Yes, there is probably a better phrasing but I want to keep "sentence" because it's like a technical term for GDevelop and I think it's more important that users to understand where it's used rather than what should be filled.